plugins/cpu/fu-cpu-device.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2019 Mario Limonciello <mario.limonciello@dell.com>
+ *
+ * SPDX-License-Identifier: GPL-2+
+ */
+
+#include "config.h"
+
+#include "fu-cpu-device.h"
+
+struct _FuCpuDevice {
+	FuDevice		 parent_instance;
+};
+
+G_DEFINE_TYPE (FuCpuDevice, fu_cpu_device, FU_TYPE_DEVICE)
+
+static
+void fu_cpu_device_parse_section (FuDevice *dev, const gchar *data)
+{
+	g_auto(GStrv) lines = NULL;
+
+	lines = g_strsplit (data, "\n", 0);
+	for (guint i = 0; lines[i] != NULL; i++) {
+		if (g_str_has_prefix (lines[i], "vendor_id")) {
+			g_auto(GStrv) fields = g_strsplit (lines[i], ":", -1);
+			if (fields[1] != NULL) {
+				g_strchug (fields[1]);
+				fu_device_set_vendor (dev, fields[1]);
+			}
+			continue;
+		}
+		if (g_str_has_prefix (lines[i], "model name")) {
+			g_auto(GStrv) fields = g_strsplit (lines[i], ":", -1);
+			if (fields[1] != NULL) {
+				g_strchug (fields[1]);
+				fu_device_set_name (dev, fields[1]);
```

Comment:
If it helps, I think you can do `fu_device_set_name (dev, g_strchug (fields[1]));`